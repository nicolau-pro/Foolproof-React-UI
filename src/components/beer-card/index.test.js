import React from 'react';
import ReactDOM from 'react-dom';
import Beercard from './index';

describe('BeerCard', () => {
  it('should be defined', () => {
    expect(Beercard).toBeDefined();
  });

  const item = {
    image_url: 'https://images.punkapi.com/v2/keg.png',
    name: 'Berliner Weisse With Yuzu - B-Sides',
    brewers_tips: 'Clean everything twice. All you want is the clean sourness of lactobacillus.',
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Beercard image={item.image_url} title={item.name} description={item.brewers_tips} />, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});

// A lot more tests could have been performed by simulating click actions on the Like button
