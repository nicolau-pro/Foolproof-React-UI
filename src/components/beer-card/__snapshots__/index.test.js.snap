// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCard renders without crashing 1`] = `
<div>
  <div
    class="sc-bZQynM hAXOBD"
  >
    <div
      class="sc-htoDjs lnaoXq"
    >
      <div
        class="sc-bxivhb kZOLLW"
      >
        <h3
          class="sc-bwzfXH hCaboq"
        >
          Berliner Weisse With Yuzu - B-Sides
        </h3>
        <p
          class="sc-htpNat hlvQJl"
        >
          Clean everything twice. All you want is the clean sourness of lactobacillus.
        </p>
      </div>
    </div>
    <div
      class="sc-gzVnrw cGNayq"
    >
      <button
        class="sc-ifAKCX ddkBQF"
      >
        <svg
          class="sc-EHOje cJrdBJ"
        >
          <path
            d="M16,28.261c0,0-14-7.926-14-17.046c0-9.356,13.159-10.399,14-0.454c1.011-9.938,14-8.903,14,0.454
 C30,20.335,16,28.261,16,28.261z"
            id="heart-icon"
          />
        </svg>
      </button>
      <img
        class="sc-bdVaJa keeMeq"
        src="https://images.punkapi.com/v2/keg.png"
      />
    </div>
  </div>
</div>
`;
